### PR TITLE
push-based messaging

### PR DESCRIPTION
**Pull request by matthiasgeihs**
Proposes to change to push-based messaging (instead of the currently used poll-based messaging) for more performance.

Depends on `MapQueue` from `@toruslabs/tss-client-util` which needs to be released first.

### CODE REVIEW

File: packages/tss-client/src/client.ts [hunk 1]
```diff
@@ -23,30 +26,7 @@ if (globalThis.js_read_msg === undefined) {
...
-            // with Error instead of string.
-            //
-            // eslint-disable-next-line prefer-promise-reject-errors
-            reject("Message not received in a reasonable time");
-          }
-          counter++;
-        }, 10);
-      });
-    }
+    const mm = await tss_client.popMessage(party, self_index, msg_type);
```
Comment (metalurgical) [inline packages/tss-client/src/client.ts:29]: Does popMessage reject if it is not found within the timeframe?

**Comment by metalurgical** [comment]
This looks good, will approve once CI passes.

**Comment by matthiasgeihs** [comment]
we will need the [@toruslabs/tss-client-util](https://github.com/torusresearch/tss-server/tree/alpha/packages/client-util) to be published for this. @himanshuchawla009 can you help?

**Comment by metalurgical** [comment]
Looks good, there's some vulnerability fixes that can be applied (see npm audit)

**Comment by matthiasgeihs** [comment]
@himanshuchawla009 what do you think?

**Review by himanshuchawla009 — APPROVED** [review]
lgtm

**Pull request merged**